Add text extraction based on ToUnicode cmap

dkaluza opened the pull request:
Fixes #125.

Heinenen commented on src/encodings/mod.rs:
```diff
@@ -1,22 +1,82 @@
+            Self::OneByteEncoding(map) => string_to_bytes(map, text),
+            Self::SimpleEncoding(name) if ["UniGB-UCS2-H", "UniGB-UTF16-H"].contains(name) => encode_utf16_be(text),
+            Self::UnicodeMapEncoding(_unicode_map) => {
+                //maybe only possible if the unicode map is an identity?
```
should be `// maybe ...`  (with space)

Heinenen reviewed (changes requested):
Some small things I would change, but overall good work!

dkaluza commented on src/encodings/mappings.rs:
```diff
@@ -1,7 +1,7 @@
 
-/// MacRomanEncoding
-pub const MAC_ROMAN_ENCODING: [Option<u16>; 256] = [
+pub type ByteToGlyphMap = [Option<u16>; 256];
```
CodedCharacterSet sound good in my opinion, changing!

dkaluza commented on src/document.rs:
```diff
@@ -570,38 +569,20 @@ impl Document {
+            string_to_bytes(&encodings::STANDARD_ENCODING, text)
         }
     }
 }
```
I agree, according to spec the StandardEncoding is a fallback only for Type 1 fonts in only certain circumstances, and is rarely used.  By leaving it I wanted to maintain current behavior.

Anyway it will be cleaned up if we remove the Option.

dkaluza commented on src/encodings/mod.rs:
```diff
@@ -1,22 +1,82 @@
+                    .collect();
+                Ok(UTF_16BE.decode(&utf16_str).0.to_string())
+            }
+            _ => Err(Error::ContentDecode),
```
Good catch, changed the code to avoid catch-all.

dkaluza commented:
Thanks for the review!
Addressed mentioned issues.

Change of the decode/encode interface resulted in some changes in parser_aux extract and replace text.
Added warnings there to inform the user that results might not be as expected in situations with None  encoding.
